File: apps/frontend/packages/api-services/src/forms/infrastructure/input/request/CreateFormRequest.ts
export default class CreateFormRequest {
	constructor(
		public readonly id: string,
		public readonly name: string,
		public readonly header: string,
		public readonly description: string,
		public readonly inputPlaceholder: string,
		public readonly buttonText: string,
		public readonly buttonColor: string,
		public readonly backgroundColor: string,
		public readonly textColor: string,
		public readonly buttonTextColor: string
	) {
    this.id = id;
    this.name = name;
    this.header = header;
    this.description = description;
    this.inputPlaceholder = inputPlaceholder;
    this.buttonText = buttonText;
    this.buttonColor = buttonColor;
    this.backgroundColor = backgroundColor;
    this.textColor = textColor;
    this.buttonTextColor = buttonTextColor;
  }
}

File: apps/backend/src/main/kotlin/com/lyra/app/forms/infrastructure/http/request/CreateFormRequest.kt
package com.lyra.app.forms.infrastructure.http.request

import jakarta.validation.constraints.NotBlank

data class CreateFormRequest(
    @field:NotBlank(message = "Name is required")
    val name: String,
    @field:NotBlank(message = "Header is required")
    val header: String,
    @field:NotBlank(message = "Description is required")
    val description: String,
    @field:NotBlank(message = "Input placeholder is required")
    val inputPlaceholder: String,
    @field:NotBlank(message = "Button text is required")
    val buttonText: String,
    @field:NotBlank(message = "Button color is required")
    val buttonColor: String,
    @field:NotBlank(message = "Background color is required")
    val backgroundColor: String,
    @field:NotBlank(message = "Text color is required")
    val textColor: String,
    @field:NotBlank(message = "Button text color is required")
    val buttonTextColor: String
)
